zircon.core/src/commonMain/kotlin/org/hexworks/zircon/api/builder/fragment/ColorThemeSelectorBuilder.kt

```diff
@@ -0,0 +1,52 @@
+package org.hexworks.zircon.api.builder.fragment
+
+import org.hexworks.zircon.api.Fragments
+import org.hexworks.zircon.api.behavior.Themeable
+import org.hexworks.zircon.api.builder.Builder
+import org.hexworks.zircon.api.data.Position
+import org.hexworks.zircon.api.fragment.MultiSelect
+import org.hexworks.zircon.api.fragment.builder.FragmentBuilder
+import org.hexworks.zircon.internal.resource.ColorThemeResource
+import kotlin.jvm.JvmStatic
+
+// constructor could be private but java can not handle private vararg constructors
+class ColorThemeSelectorBuilder(
+        width: Int,
+        vararg componentsToUpdate: Themeable
+) : FragmentBuilder<MultiSelect<ColorThemeResource>, ColorThemeSelectorBuilder> {
+
+    companion object {
+        @JvmStatic
+        fun newBuilder(width: Int, vararg componentsToUpdate: Themeable) = ColorThemeSelectorBuilder(width, *componentsToUpdate)
+    }
+
+    private val multiSelectBuilder: MultiSelectBuilder<ColorThemeResource>
+
+    init {
+        require(componentsToUpdate.isNotEmpty()) {
+            "ColorThemeSelector needs a list of components it can update. Supplied list was empty."
+        }
+        val allColorThemes: List<ColorThemeResource> = ColorThemeResource
+                .values()
+                .sortedBy { it.name }
+        multiSelectBuilder = Fragments
+                .multiSelect(width, allColorThemes)
+                .withCallback { _, newThemeResource -> componentsToUpdate.forEach { it.themeProperty.updateValue(newThemeResource.getTheme()) } }
```

Comment:
I'm gonna update `MultiSelect` so that it will have a `selected` `Property` and then we can just use data binding instead of a callback. If we're using a `Group` for the components we won't need the loop either. In its final form this will look like: `group.theme.updateFrom(themeSelector.theme)` or something similar.